src/utils.ts

```diff
@@ -1,85 +1,85 @@
 import * as cheerio from "cheerio";
-import { Err, Ok, Result } from "./graduate-types/common";
-import { existsSync } from "fs";
-import { mkdir, readFile, writeFile } from "fs/promises";
+import { Err, Ok, type Result } from "./graduate-types/common";
 import undici from "undici";
 
+export const loadHTML = async (url: string): Promise<CheerioStatic> => {
+  return cheerio.load(await wrappedGetRequest(url));
+};
+
 export const loadHtmlWithUrl = async (
   url: URL,
 ): Promise<{ url: URL; result: Result<CheerioStatic, unknown> }> => {
   let result: Result<CheerioStatic, unknown>;
   try {
-    result = Ok(await loadHTML(url.href));
+    const html = cheerio.load(await wrappedGetRequest(url.href));
+    result = Ok(html);
   } catch (error) {
     result = Err(error);
   }
   return { url, result };
 };
 
-/**
- * Whether to cache the response bodies of requests. if set to true,
- * `cachedGetRequest` will cache requests.
- */
-const USE_CACHE = true;
-export const loadHTML = async (url: string): Promise<CheerioStatic> => {
-  const data = await cachedGetRequest(url);
-  return cheerio.load(data);
-};
-
-/**
- * If use cache is true, will attempt to look for request body in
- * `./catalogCache` before fetching. If does not exist, will save the response
- * in `./catalogCache` before returning response.
- *
- * @param url
- */
-const cachedGetRequest = async (url: string) => {
-  if (!USE_CACHE) {
-    return await wrappedGetRequest(url);
-  }
-
-  if (!existsSync("./catalogCache")) {
-    await mkdir("./catalogCache");
-  }
-
-  // https://stackoverflow.com/questions/35511331/how-to-make-a-valid-filename-from-an-arbitrary-string-in-javascript
-  const path = `./catalogCache/${url.replaceAll(/[\/|\\:*?"<>]/g, "-")}.html`;
-  if (existsSync(path)) {
-    return await readFile(path);
-  }
-
-  const data = await wrappedGetRequest(url);
-  await writeFile(path, data);
-  return data;
-};
-
 export const wrappedGetRequest = async (url: string) => {
   const response = await undici.request(url, { maxRedirections: 1 });
   if (response.statusCode !== 200) {
-    throw new Error(`non-ok status code: ${response.statusCode}, url: ${url}`);
+    throw new Error(`Non-ok status code: ${response.statusCode}, url: ${url}`);
   }
   return await response.body.text();
 };
 
-export const ensureLength = <T>(n: number, l: T[], extraMessage?: string) => {
-  const length = l.length;
-  if (length !== n) {
-    const msg = `expected array length to equal exactly ${n}, but was ${length}${
-      extraMessage ?? ""
-    }`;
-    throw new Error(msg);
+export function ensureExactLength<T, N extends number>(
+  arr: Array<T>,
+  length: N,
+  errorMessage?: string,
```

Comment:
Hm.... might be worth writing this method (and other helper methods) to accept objects as parameters to improve readability (and if needed, extensibility idk) (this is something I've seen my manager do a bit for those reasons, so thought i'd mention it lol)

So we could do something like 

```
export function ensureExactLength<T, N extends number>({
  array: Array<T>,
  length: N,
  errorMessage?: string,
}: EnsureLengthInputs) => {...}
```

and then calling it:

```
ensureExactLength({
  array: tabsArr,
  length: 2,
  errorMessage: 'this is an error message'
});
```

not to mention these three methods all seem to accept the same parameters

Reply:
This makes senses, but I would argue for this case it just add more boilerplate since the number of arguments is not that high (> 3), and the function is expressive enough that you don't need to go to the function definition to see what it is doing (`ensureExactLength` is pretty self-explanatory). This is mainly based on this [stackoverflow](https://stackoverflow.com/questions/12826977/multiple-arguments-vs-options-object)